logstash-core/lib/logstash/filter_delegator.rb

```diff
@@ -37,7 +37,9 @@ def config_name
     def multi_filter(events)
       @metric_events.increment(:in, events.size)
 
+      start = Time.now
       new_events = @filter.multi_filter(events)
+      @filter.metric.increment(:time_ms, ((Time.now - start)*1000).to_i)
```

Comment:
@ph regarding @metric_events.time { ... }, method call overhead is high in Ruby/JRuby, and my experience has me try to to avoid proc/lambda blocks in performance-needing code paths. For future task, maybe we can have a Stopwatch class, something like:

```ruby
clock = @metric_events.time
@filter.multi_filter(events)
clock.stop # Compute duration and increment metric_events metric.
```